sharding/xorValidatorShuffler.go

```diff
@@ -102,20 +102,28 @@ func (rxs *randXORShuffler) UpdateNodeLists(args ArgsUpdateNodes) (map[uint32][]
 }
 
 func removeFromWaiting(waitingNodes map[uint32][]Validator, leavingNodes []Validator) (map[uint32][]Validator, []Validator) {
-	for shard, vList := range waitingNodes {
-		nbToRemove := len(vList)
-		if len(leavingNodes) < nbToRemove {
-			nbToRemove = len(leavingNodes)
-		}
-
-		vList, removedNodes := removeValidatorsFromList(vList, leavingNodes, nbToRemove)
-		leavingNodes, _ = removeValidatorsFromList(leavingNodes, removedNodes, len(removedNodes))
-		waitingNodes[shard] = vList
+	for shard := uint32(0); shard < uint32(len(waitingNodes)-1); shard++ {
```

Comment:
this won't work as expected, and might panic, if we decide that we will call this function on a subset of all shard's validators. Why not take the available keys, sort them and then iterate on them?

Reply:
good catch. there are multiple places in the application where we do a range over the maps and expect that the number of shards is equal to the len of the maps.

Reply:
not the case any more. using: sort keys on map